iroh-rpc/src/client.rs

```diff
@@ -254,59 +259,122 @@ mod test {
         let a_addr: Multiaddr = "/memory/1234".parse().unwrap();
-
-        let b_state = handler::State::new("Wooo!".to_string());
         let b_addr: Multiaddr = "/memory/4321".parse().unwrap();
         let b_keypair = Keypair::generate_ed25519();
         let b_peer_id = b_keypair.public().to_peer_id();
-
-        let (mut a_client, a_server) = rpc_from_config(
-            RpcConfig::new()
-                .with_swarm(swarm::new_mem_swarm(a_keypair))
-                .with_state(a_state)
-                // can add the addrs here, or after the client has already
-                // been constructed
-                .with_addr("b", b_addr.clone(), b_peer_id),
+        let a_keypair = Keypair::generate_ed25519();
+        let a_peer_id = a_keypair.public().to_peer_id();
+        client_example(
+            TestConfig {
+                addr: a_addr,
+                peer_id: a_peer_id,
+                swarm: swarm::new_mem_swarm(a_keypair),
+            },
+            TestConfig {
+                addr: b_addr,
+                peer_id: b_peer_id,
+                swarm: swarm::new_mem_swarm(b_keypair),
+            },
         )
-        .expect("rpc client to be created");
-        let (mut b_client, b_server) = rpc_from_config(
-            RpcConfig::new()
-                .with_swarm(swarm::new_mem_swarm(b_keypair))
-                .with_state(b_state)
-                .with_namespace("b", |n| n.with_method("get", get))
-                // can add the addrs here, or after the client has already
-                // been constructed
-                .with_addr("a", a_addr.clone(), a_peer_id),
+        .await;
+    }
+
+    #[tokio::test]
+    async fn tcp_client_example() {
+        let addr: Multiaddr = "/ip4/0.0.0.0/tcp/0".parse().unwrap();
+        let b_keypair = Keypair::generate_ed25519();
+        let b_peer_id = b_keypair.public().to_peer_id();
+        let a_keypair = Keypair::generate_ed25519();
+        let a_peer_id = a_keypair.public().to_peer_id();
+        client_example(
+            TestConfig {
+                addr: addr.clone(),
+                peer_id: a_peer_id,
+                swarm: swarm::new_swarm(a_keypair)
+                    .await
+                    .expect("swarm build failed"),
+            },
+            TestConfig {
+                addr,
+                peer_id: b_peer_id,
+                swarm: swarm::new_swarm(b_keypair)
+                    .await
+                    .expect("swarm build failed"),
+            },
         )
-        .expect("rpc client to be created");
+        .await;
+    }
+
+    async fn client_example(a: TestConfig, b: TestConfig) {
+        let a_state = handler::State::new(());
+        let b_state = handler::State::new("Wooo!".to_string());
+
+        let (mut a_client, a_server) = RpcBuilder::new()
+            .with_swarm(a.swarm)
+            .with_state(a_state)
+            .with_namespace("a", |n| n.with_method("ping", pong))
```

Comment:
should the be `pong` or `ping?

Reply:
Great question. Probably whatever convention we want to give it. My idea here was that `a` sends `ping` and `b`, that this is getting set on sends back `pong`.

But it's confusing. So yea, let's say make a convention that the methods are about handling the request, not about what gets sent back.